homeassistant/components/rainbird/quality_scale.yaml

```diff
@@ -0,0 +1,85 @@
+rules:
+  # Bronze
+  config-flow:
+    status: todo
+    comment: Some fields are missing data descriptions.
+  brands: done
+  dependency-transparency: done
+  common-modules: done
+  has-entity-name: done
+  action-setup:
+    status: done
+    comment: |
+      The integration only has an entity service, registered in the platform.
+  appropriate-polling:
+    status: done
+    comment: |
+      Rainbird devices are local. Irrigation valve/controller status is polled
+      once per minute to get fast updates when turning on/off the valves.
+      The irrigation schedule uses a 15 minute poll interval since it rarely
+      changes.
+
+      Rainbird devices can only accept a single http connection, so this uses a
+      an aiohttp.ClientSession with a connection limit, and also uses a request
+      debouncer.
+  test-before-configure: done
+  entity-event-setup:
+    status: exempt
+    comment: Integration is polling and does not subscribe to events.
+  unique-config-entry: done
```

Comment:
Why is the unique_id nullable? When does that happen and can we fallback to the entry id when that happens?

Reply:
Oh I meant to post this under `entity-unique-id`

Reply:
Only in old migration cases:

https://github.com/home-assistant/core/blob/502a221feb345ce434e265be5dcfb44176828950/homeassistant/components/rainbird/config_flow.py#L177

Reply:
Also used to be configurable by yaml

Reply:
Should we fallback to entry_id when that happens?